### PR TITLE
Upgrade MIME types, fix Model.js to not look at URL file extension.

### PR DESCRIPTION
**Pull request by emackey**
This addresses @mramato's comment in 3a5eb8551, and also fixes a bug in `Model.js` where the URL itself was being used to distinguish `bgltf` from `gltf`.  Using the URL is incorrect because data-URIs do not contain file extensions.  Likewise, models fetched from a server-side database may not contain a filename with an extension.  So instead, we look at the binary magic number to see which path to take.

Thanks to @shunter for research & advice on this.  We should probably get his approval before merge.

### CODE REVIEW

**Comment by mramato** [comment]
:+1:

**Comment by pjcozzi** [comment]
OK with me.  The only concern is if `getStringFromTypedArray` is a performance hit when `TextDecoder` isn't available, but that is on the slow path anyway, i.e., non Binary glTF.

**Comment by emackey** [comment]
It is a small performance hit, but only on the "slow path" as you mention.  The problem is the XHR responseType can only be set before making the request, and Cesium won't know if it's getting binary or JSON until after the request is made and the response starts to appear.  Expecting known file extensions to show up in 3rd-party app URLs is not the correct way to guess the outcome.  So instead, we expect binary, and take steps to convert back to a string if needed.

**Comment by pjcozzi** [comment]
@shunter did you want to look at this?

File: Source/Scene/Model.js [hunk 1]
```diff
@@ -800,6 +802,13 @@ define([
         options.cacheKey = cacheKey;
         var model = new Model(options);
 
+        if (!defined(options.headers)) {
+            options.headers = {};
+        }
+        if (!defined(options.headers.Accept)) {
```
Comment (shunter) [inline Source/Scene/Model.js]: the call to `clone` above needs to be changed to pass `true` to deep clone, since we are modifying a sub-object now.

Comment (emackey) [inline Source/Scene/Model.js]: Thanks, done.

File: Source/Scene/Model.js [hunk 2]
```diff
@@ -810,17 +819,17 @@ define([
             setCachedGltf(model, cachedGltf);
             gltfCache[cacheKey] = cachedGltf;
 
-            if (/\.bgltf$/i.test(url)) {
-                // Load binary glTF
-                loadArrayBuffer(url, options.headers).then(function(arrayBuffer) {
+            loadArrayBuffer(url, options.headers).then(function(arrayBuffer) {
+                var magic = getStringFromTypedArray(arrayBuffer, 0, 4);
```
Comment (shunter) [inline Source/Scene/Model.js]: This will crash if arrayBuffer is less than 4 bytes.  For example, try:

`getStringFromTypedArray(new ArrayBuffer(), 0, 4)`

Comment (emackey) [inline Source/Scene/Model.js]: Can't fit any valid GLTF JSON in 3 bytes or less.  Should we explicitly throw RuntimeError?

Comment (emackey) [inline Source/Scene/Model.js]: `JSON.parse` will throw on an empty string too.

Comment (shunter) [inline Source/Scene/Model.js]: Just parse as JSON as it did before, and let it do the existing checks.

Comment (pjcozzi) [inline Source/Scene/Model.js]: `DeveloperError` like the other I suppose.

Comment (pjcozzi) [inline Source/Scene/Model.js]: or just do what @shunter says.

Comment (emackey) [inline Source/Scene/Model.js]: Ok.  How's this?

**Pull request merged**